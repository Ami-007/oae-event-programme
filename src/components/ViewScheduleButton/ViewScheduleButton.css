@import"https://fonts.googleapis.com/css2?family=PT+Sans&family=Poppins:wght@400;500;600;700&display=swap";@import"https://fonts.googleapis.com/css2?family=Lato&family=Playfair+Display:wght@400;500;600;700&display=swap";.view-schedule-button{height:66px;width:335px;text-transform:uppercase;background-color:#3c3c3c;border-radius:5px;color:#fff;font-weight:700;font-family:"Poppins",sans-serif;font-size:1rem;display:flex;flex-direction:column;justify-content:center;align-items:center;text-decoration:none}