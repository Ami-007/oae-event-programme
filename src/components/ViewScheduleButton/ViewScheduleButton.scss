@use "../../assets/sass/variables.scss" as *;
@use "../../assets/sass/typography.scss" as *;

.view-schedule-button {
  height: 66px;
  width: 335px;
  text-transform: uppercase;
  background-color: $btn-color;
  border-radius: 5px;
  color: $color-white;
  font-weight: 700;
  font-family: $modern-header;
  font-size: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  margin: auto;
}
