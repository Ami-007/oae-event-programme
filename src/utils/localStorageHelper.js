export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
