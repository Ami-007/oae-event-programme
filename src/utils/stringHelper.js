export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
