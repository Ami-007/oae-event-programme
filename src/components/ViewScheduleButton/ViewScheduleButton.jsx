import React from "react";
import "./ViewScheduleButton.scss";
import { Link } from "react-router-dom";

const ViewScheduleButton = () => {
  return (
    <Link to="/schedule">
      <button className="schedule-button">View Schedule</button>
    </Link>
  );
};
export default ViewScheduleButton;
