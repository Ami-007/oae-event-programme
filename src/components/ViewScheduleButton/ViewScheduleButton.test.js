import { screen } from "@testing-library/react";
import ViewScheduleButton from "./ViewScheduleButton";
import { customRender } from "../../utils/testUtils";

it("should render the button", () => {
  customRender(<ViewScheduleButton eventId={1234}/>);
  const button = screen.getByRole("link");

  expect(button).toMatchSnapshot();
});