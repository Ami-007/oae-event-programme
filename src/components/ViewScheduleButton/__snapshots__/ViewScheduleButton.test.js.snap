// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the button 1`] = `
<a
  class="view-schedule-button"
  href="/event-schedule"
>
  View Schedule
</a>
`;
